api-server/scanner-lib/src/blockchain_state/mod.rs

```diff
@@ -99,6 +100,16 @@ impl<S: ApiServerStorage + Send + Sync> LocalBlockchainState for BlockchainState
         Ok(best_block)
     }
 
+    async fn update_mempool_feerate_points(
+        &mut self,
+        feerate_points: Vec<(usize, FeeRate)>,
+    ) -> Result<(), Self::Error> {
+        let mut db_tx = self.storage.transaction_rw().await.expect("Unable to connect to database");
+        db_tx.set_feerate_points(feerate_points).await?;
```

Comment:
Why would you need the feerate points to be stored in the database at all? You just need some simple `Mutex<(Time,Points)>` on a value in your state. If current time minus Time stored there is larger than X, then update the cached value.

Reply:
fixed to use in memory caching state